src/isar/state_machine/state_machine.py

```diff
@@ -321,8 +286,39 @@ def _check_dependencies(self):
                 return False
         return True
 
+    def publish_telemetry_thread(self) -> None:
+        robot_id = settings.ROBOT_ID
+        PUBLISH_FREQUENCY = 1
+
+        messenger = Telemetry()
+
+        while True:
+            timestamp = datetime.now().strftime("%Y-%m-%dT%H:%M:%SZ")
+
+            state: IsarStatePayload.State = IsarStatePayload.State(self.current_state)
+            payload_state = IsarStatePayload(state=state, timestamp=timestamp).to_json()
+
+            battery_payload: int = 100
+            payload_battery = TelemetryBatteryPayload(
+                battery=battery_payload, timestamp=timestamp
+            ).to_json()
+
+            pose: Pose = self.robot.get_pose()
+            euler_orientation: np.ndarray = pose.orientation.to_euler_array()
+            pose_payload: TelemetryPosePayload.Pose = TelemetryPosePayload.Pose(
+                x=pose.position.x,
+                y=pose.position.y,
+                z=pose.position.z,
+                yaw=euler_orientation[0],
+                pitch=euler_orientation[1],
+                roll=euler_orientation[2],
+            )
+            payload_pose = TelemetryPosePayload(
+                pose=pose_payload, timestamp=timestamp
+            ).to_json()
+
+            messenger.publish(f"isar/{robot_id}/state", payload_state)
```

Comment:
Should the state be published at a second interval? It's a timeseries that only changes once in a while and it would be better to publish on change rather than continuously.

Reply:
Another reason for publishing on changes is that then we can treat it like an event. If we publish every second Flotilla will have to make queries to the database every second for other topics that will show the current task status. Thus we need to support event based topics in some way which isn't possible with the current thread framework.

Reply:
We should definitely treat it as an event and not like time series data.